service-base/src/main/java/org/eclipse/hono/service/http/HttpServiceBase.java

```diff
@@ -0,0 +1,461 @@
+     * <p>
+     * Subclasses may override this method to do any kind of initialization work.
+     *
+     * @return A future indicating the outcome of the operation. The start up process fails if the returned future
+     *         fails.
+     */
+    protected Future<Void> preStartServers() {
+        return Future.succeededFuture();
+    }
+
+    @Override
+    protected final Future<Void> startInternal() {
+
+        return preStartServers()
+                .compose(s -> checkPortConfiguration())
+                .compose(s -> {
+                    Router router = createRouter();
+                    if (router == null) {
+                        return Future.failedFuture("no router configured");
+                    } else {
+                        addRoutes(router);
+                        return CompositeFuture.all(bindSecureHttpServer(router), bindInsecureHttpServer(router));
+                    }
+                })
+                .compose(s -> onStartupSuccess());
+    }
+
+    /**
+     * Invoked after the http server has been started successfully.
+     * <p>
+     * May be overridden by sub-classes.
+     *
+     * @return A future indicating the outcome of the operation. The start up process fails if the returned future
+     *         fails.
+     */
+    protected Future<Void> onStartupSuccess() {
+
+        return Future.succeededFuture();
+    }
+
+    /**
+     * Creates the router for handling requests.
+     * <p>
+     * This method creates a router instance with the following routes:
+     * <ol>
+     * <li>A default route limiting the body size of requests to the maximum payload size set in the <em>config</em>
+     * properties.</li>
+     * <li>A route for retrieving this adapter's current status from the resource path returned by
+     * {@link #getStatusResourcePath()} (if not {@code null}).</li>
+     * </ol>
+     *
+     * @return The newly created router (never {@code null}).
+     */
+    protected Router createRouter() {
+
+        final Router router = Router.router(vertx);
+        LOG.info("limiting size of inbound request body to {} bytes", getConfig().getMaxPayloadSize());
+        router.route().handler(BodyHandler.create().setBodyLimit(getConfig().getMaxPayloadSize())
+                .setUploadsDirectory(DEFAULT_UPLOADS_DIRECTORY));
+
+        String statusResourcePath = getStatusResourcePath();
+        if (statusResourcePath != null) {
+            router.route(HttpMethod.GET, statusResourcePath).handler(this::doGetStatus);
+        }
+
+        return router;
+    }
+
+    /**
+     * Returns the path for the status resource.
+     * <p>
+     * By default, this method returns {@code /status}. Subclasses may override this method to return a different path
+     * or {@code null}, in which case the status resource will be disabled.
+     *
+     * @return The resource path or {@code null}.
+     */
+    protected String getStatusResourcePath() {
```

Comment:
we should get rid of this in favor of subclasses registering a proper health check